lib/SILOptimizer/Mandatory/TFPartition.cpp

```diff
@@ -3840,7 +3840,9 @@ void TFFunctionPartition::insertTensorComputationStartEndTerminate(
     // closed over.  If it is a TensorHandle<T>, load the CTensorHandle out of
     // it.  If it is a scalar, then we need to box the scalar in a
     // CTensorHandle.
-    if (isTensorHandle(tensorValue->getType().getASTType())) {
+    if ((TFSendRecvOpaqueHandle &&
+         isTensorFlowValue(tensorValue->getType().getASTType())) ||
```

Comment:
does isTensorFlowValue() subsume isTensorHandle()?

Reply:
Yes it does: https://github.com/apple/swift/blob/fd5acc12c899d2c7462d5253d79041729c9f3d1f/lib/AST/TensorFlow.cpp#L169